File: VantaSpeech-Windows/VantaSpeech/Models/TranscriptionResult.cs
using System.Text.Json.Serialization;

namespace VantaSpeech.Models;

public class TranscriptionResult
{
    [JsonPropertyName("transcription")]
    public string Transcription { get; set; } = string.Empty;

    [JsonPropertyName("summary")]
    public string? Summary { get; set; }

    [JsonPropertyName("language")]
    public string? Language { get; set; }

    [JsonPropertyName("duration")]
    public double? Duration { get; set; }
}

public class TranscriptionError
{
    [JsonPropertyName("error")]
    public string Error { get; set; } = string.Empty;

    [JsonPropertyName("code")]
    public string? Code { get; set; }
}

File: VantaSpeech-Android/app/src/main/java/com/vantaspeech/data/model/TranscriptionResult.kt
package com.vantaspeech.data.model

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class TranscriptionResult(
    @Json(name = "transcription")
    val transcription: String,

    @Json(name = "summary")
    val summary: String? = null,

    @Json(name = "language")
    val language: String? = null,

    @Json(name = "duration")
    val duration: Double? = null
)

@JsonClass(generateAdapter = true)
data class TranscriptionError(
    @Json(name = "error")
    val error: String,

    @Json(name = "code")
    val code: String? = null
)

sealed class TranscriptionState {
    data object Idle : TranscriptionState()
    data object Uploading : TranscriptionState()
    data object Processing : TranscriptionState()
    data class Success(val result: TranscriptionResult) : TranscriptionState()
    data class Error(val message: String) : TranscriptionState()
}
